packages/node/src/behaviors/operational-credentials/OperationalCredentialsServer.ts

```diff
@@ -308,32 +308,39 @@ export class OperationalCredentialsServer extends OperationalCredentialsBehavior
 
         if (timedOp.fabricIndex !== undefined) {
             throw new StatusResponseError(
-                `updateNoc received after ${timedOp.forUpdateNoc ? "UpdateNOC" : "AddNOC"} already invoked.`,
+                `updateNoc received after ${timedOp.forUpdateNoc ? "UpdateNOC" : "AddNOC"} already invoked`,
                 StatusCode.ConstraintError,
             );
         }
 
-        if (timedOp.forUpdateNoc) {
+        if (timedOp.forUpdateNoc === false) {
             throw new StatusResponseError(
-                `addNoc received after csr request was invoked for UpdateNOC.`,
+                `UpdateNoc received after csr request was invoked for AddNOC`,
                 StatusCode.ConstraintError,
             );
         }
 
-        if (timedOp.hasRootCert) {
-            throw new StatusResponseError(
-                "Trusted root certificate added in this session which is now allowed for UpdateNOC.",
-                StatusCode.ConstraintError,
-            );
+        if (timedOp.rootCertSet) {
+            return {
+                // TC_OPCREDS_3_4 expects MissingCsr here...  Previously we used InvalidNoc which seems like it makes
+                // more sense
+                statusCode: OperationalCredentials.NodeOperationalCertStatus.MissingCsr,
```

Comment:
This is against Spec and against Test plan. So it is an SDK AND test error as it seems. Clarification ongoing

Reply:
OK, I modified to conform to specification.  See if it looks correct now, if so I'll handle in testing branch after merge.